pandas/core/dimensioned.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np
+
+from pandas.core.base import (PandasObject)
+from pandas.util._decorators import cache_readonly
+from pandas import compat
+from pandas.core.common import is_null_slice
+
+
+class Dimensional(PandasObject):
+    """
```

Comment:
Thanks for the feedback. What you said is exactly what I planned to do, but I somehow don't get it how to do it.

The problem is that `Series` does not seem to habe a dtype other than the numpy array's dtype:
If I understood the code correctly, then the following happens:

If I do `Series([1,2,3], dtype=DimensionedFloatDtype("meter"))`, then the code 
hits `data = _sanitize_array(data, index, dtype, copy, raise_cast_failure=True)` which returns an `np.array`.
This array gets passed into the SingleBlockManager which is passed to `NDFrame.__init__`, where it is stored as `_data`.

Now `Series.dtype` is a property that just retrieves the dtype from the SingleBlockManager (which was constructed with a np.array, and according to `_interleaved_dtype(blocks)` disallows ExtensionDtypes).

The way DatetimeTZDtype does it is by creating a `pandas.core.indexes.DatetimeIndex`, categorical uses `Categorical` to mimic an array.

 I'm sorry if I missed something obvious, but I cannot see how to do this without the need for a wrapper class that holds my float data.